parser.rb

```diff
@@ -0,0 +1,24 @@
+class ParserGamesLogs
+  def initialize(filename_with_path)
+    @filename_with_path = filename_with_path
+    @mode = 'r'
+    file_exists?()
+  end
+  
+
+  def read_first_line()
+    if file_exists?
```

Comment:
Se no método contrutor você ja verifica se o arquivo existe, por que verificar novamente?

Reply:
Porque o arquivo pode deixar de existir durante o tempo de vida de instância do objeto. Como o arquivo é sempre fechado após alguma operação ser realizada, ele fica livre para ser alterado por qualquer outro processo. Então ele verifica se o arquivo ainda existe quando for realizar alguma outra operação. A verificação no construtor é para impedir de a classe ser instanciada se o arquivo não existir durante a criação do objeto.

Reply:
:eyes: tendi... ta de boa, pode meter ficha